docs/spreadsheets/exercises/writing-scts.md

```diff
@@ -0,0 +1,264 @@
+this.
+
+**Note:** for this SCT, the solution is not used
+
+Example:
+
+```
+? check_regex(
+  pattern = "1\\.1\\s*-\\s*1",
+  message = "Please subtract one from `1.1`."
+)
+```
+
+#### check_function
+
+```
+check_function(
+  function = string,
+  message = string,
+  suggestion = string
+)
+```
+
+Checks whether the user used a certain function. The functions are normalized, so capitalization is
+not an issue. A more tailored feedback message will be used automatically, so this function can
+often be used without any specific `message` or `suggestion`.
+
+**Note:** for this SCT, the solution is not used
+
+Example:
+
+```
+? check_function(function = "SUM")
+```
+
+#### check_operator
+
+```
+check_operator(
+  operator = string,
+  message = string,
+  suggestion = string
+```
+
+Checks whether the user used a certain operator. A more tailored feedback message will be used
+automatically, so this function can often be used without any specific `message` or `suggestion`.
+
+**Note:** for this SCT, the solution is not used
+
+Example:
+
+```
+? check_operator(operator = "+")
+```
+#### check_reference
+
+```
+check_reference(
+  reference = string,
+  message = string,
+  suggestion = string
+)
+```
+
+Checks whether the user used a certain reference. A more tailored feedback message will be used
+automatically, so this function can often be used without any specific `message` or `suggestion`.
+
+
+**Note:** for this SCT, the solution is not used. However, this function can often be replaced by
+the more convenient `check_references` or `check_absolute_references`, which **do** look at the
+solution.
+
+Examples:
+
+```
+? check_reference(reference = "B3")
+```
+
+```
+? check_reference(reference = "$A$27")
+```
+
+
+#### check_references
+
+```
+check_references(
+  message = string,
+  suggestion = string
+)
+```
+
+Similarly to `check_reference`, check whether the correct references are used in a cell. However,
+this function will compare the user's input to the solution. It will check whether all references
+that are used in the solution, are also used in the user's input. Does not match absolute
+references. It can often be used without arguments.
+
+Example:
+
+```
+? check_references
+```
+
+
+#### check_absolute_references
+
+```
+check_absolute_references(
+  message = string,
+  suggestion = string
+)
+```
+
+Does exactly the same as `check_references`, but for absolute references.
+
+Example:
+
+```
+? check_absolute_references
+```
+
+#### check_correct
+
+```
+check_correct(
+  check = string,
+  diagnose = string
+)
+```
+
+This is the only **meta** SCT function. `check_correct` accepts 2 arguments: `check` and `diagnose`
+and both of them are SCTs themselves.
+
+The function will first check whether the SCT in `check` passes. If it passes, the SCT in `diagnose`
+is never executed and the SCT passes. If `check` fails, `diagnose` is ran and will generally give
```

Comment:
Can you recursively nest calls to `check_correct()`. As in

```
? check_correct(
  check = check_value,
  diagnose = {
    check_correct(
      check = check_function(function = "SUM")
      diagnose = check_reference
    );
  }
)
```

Reply:
Yes, you could use that. The syntax you proposed is slightly off though:
```
? check_correct(
  check = check_value,
  diagnose = {
    check_correct(
      check = check_function(function = "SUM")
      diagnose = check_reference
    );
  }
)
```

And you could simplify this to:

```
? check_correct(
  check = check_value,
  diagnose = check_correct(
    check = check_function(function = "SUM"),
    diagnose = check_references
  )
)
```